padertorch/data/fragmenter.py

```diff
@@ -15,110 +16,199 @@
-            {key: example[key] for key in self.copy_keys}
-            if self.copy_keys is not None else example
-        )
 
-        if random_onset:
-            start = np.random.rand()
-            for fragment_step in self.fragment_steps.values():
-                start = int(int(start*fragment_step) / fragment_step)
+    def __init__(self, length: int = -1,shift: int = None,
+                 include_keys: Union[str, list, tuple] = None,
+                 exclude_keys: Union[str, list, tuple] = None,
+                 copy_keys: Union[str, list, tuple] = None,
+                 axis: Union[int, list, tuple, dict] = -1,
+                 anchor_mode: str = 'left', rng=np.random):
+
+        self.include = None if include_keys is None else to_list(include_keys)
+        self.exclude = [] if exclude_keys is None else to_list(exclude_keys)
+        self.length = length
+        if isinstance(axis, (dict, int)):
+            self. axis = axis
+        elif isinstance(axis, (tuple, list)):
+            self.axis = to_list(axis)
+            assert len(axis) == len(include_keys), (
+                'If axis are specified as list it has to have the same length'
+                'as include_keys', axis, include_keys
+            )
         else:
-            start = 0.
-
-        def fragment(key, x):
-            fragment_step = self.fragment_steps[key]
-            fragment_length = self.fragment_lengths[key]
-            start_idx = int(start * fragment_step)
-            if start_idx > 0:
-                slc = [slice(None)] * len(x.shape)
-                slc[self.axis] = slice(
-                    int(start_idx), x.shape[self.axis]
-                )
-                x = x[slc]
-
-            end_index = x.shape[self.axis]
-            if self.drop_last:
-                end_index -= (fragment_length - 1)
-            fragments = list()
-            for start_idx in np.arange(0, end_index, fragment_step):
-                if fragment_length == 1 and self.squeeze:
-                    fragments.append(x.take(start_idx, axis=self.axis))
-                else:
-                    slc = [slice(None)] * len(x.shape)
-                    slc[self.axis] = slice(
-                        int(start_idx), int(start_idx) + int(fragment_length)
-                    )
-                    fragments.append(x[tuple(slc)])
-            return fragments
-
-        features = flatten({
-            key: nested_op(lambda x: fragment(key, x), example[key])
-            for key in self.fragment_steps.keys()
-        })
-        num_fragments = np.array(
-            [len(features[key]) for key in list(features.keys())]
+            raise TypeError('Unknown type for axis', axis)
+        self.shift = shift
+        self.anchor_mode = anchor_mode
+        self.rng = rng
+        self.copy_keys = None if copy_keys is None else to_list(copy_keys)
+
+    def __call__(self, example):
+        # Shortcut if segmentation is disabled
+        if self.length == -1:
```

Comment:
Replace this with `if len(self.include) == 0:`

Reply:
I wouldn't replace it. You might want to disable chunking from the config and then it is much easier to set the length to -1 than to change the include keys.